null-payment-plugin/src/payments/payment_callbacks.rs

```diff
@@ -0,0 +1,395 @@
+extern crate libc;
+extern crate rand;
+
+use indy::api::payments::indy_register_payment_method;
+use indy::api::ledger::indy_build_get_txn_request;
+use indy::api::ErrorCode;
+use self::rand::Rng;
+use std::sync::Mutex;
+use std::sync::atomic::Ordering;
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::ATOMIC_USIZE_INIT;
+use std::collections::HashMap;
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::ffi::CStr;
+
+type CommonResponseCallback = extern fn(command_handle_: i32,
+                                        err: ErrorCode,
+                                        res1: *const c_char) -> ErrorCode;
+
+lazy_static! {
+    static ref IDS_COUNTER: AtomicUsize = ATOMIC_USIZE_INIT; //TODO use AtomicI32
+}
+
+pub extern fn nullpayment_init(cb: Option<extern fn(cmd_handle: i32, err: ErrorCode)>) {
+    let _cmd_handle = get_next_id();
```

Comment:
Why we generate new cmd_handle here instead of pass user's value directly?

Reply:
Fixed this